### PR TITLE
Record-inize methods

### PR DESCRIPTION
**Pull request by NotMyFault**
This is an alternative approach to https://github.com/IntellectualSites/PlotSquared/pull/3942/, without deprecations in v6 because the methods are subject to removal anyway. Therefore, deprecating them pointing to the drop-in methods is a bit _point_less.

I don't plan to spend more time implementing static accessors in v6 at this point.

### CODE REVIEW

**Comment by NotMyFault** [comment]
@SirYwell Your favorite checkstyle has been applied.

File: Core/src/main/java/com/plotsquared/core/plot/BlockBucket.java [hunk 1]
```diff
@@ -215,66 +215,48 @@ public int hashCode() {
...
-        public int getMax() {
-            return this.max;
-        }
+            public boolean isInRange(final int num) {
+                return num <= max && num >= min;
+            }
 
-        public boolean equals(final Object o) {
-            if (o == this) {
+            public boolean equals(final Object o) {
```
Comment (SirYwell) [inline Core/src/main/java/com/plotsquared/core/plot/BlockBucket.java:232]: I think we can fully remove the equals and hashCode methods here

File: Core/src/main/java/com/plotsquared/core/services/plots/AutoQuery.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    /**
+     * Crate a new auto query
+     *
+     * @param player   Player to claim for
+     * @param startId  Plot ID to start searching from
+     * @param sizeX    Number of plots along the X axis
+     * @param sizeZ    Number of plots along the Z axis
+     * @param plotArea Plot area to search in
+     */
+    public AutoQuery(
```
Comment (SirYwell) [inline Core/src/main/java/com/plotsquared/core/services/plots/AutoQuery.java:44]: Annotations can be added to the record components, and the `@param` tag is allowed at the record header too. Can we remove the constructor as well as the methods?

File: Core/src/main/java/com/plotsquared/core/uuid/UUIDMapping.java [hunk 3]
```diff
@@ -20,29 +20,15 @@
...
-    public @NonNull UUID getUuid() {
-        return this.uuid;
-    }
+public record UUIDMapping(
+        @NonNull UUID uuid,
+        @NonNull String username
+) {
 
     @Override
     public boolean equals(final Object o) {
```
Comment (SirYwell) [inline Core/src/main/java/com/plotsquared/core/uuid/UUIDMapping.java:34]: This method can be dropped too I guess?

**Review by SirYwell — CHANGES_REQUESTED** [review]
There are a few places were we can probably remove even more code, but looks nice!

**Review by SirYwell — APPROVED** [review]

**Pull request merged**